### PR TITLE
[WIP] Types: remove chainer interface fallback

### LINKED ISSUE #5573: Suggestion: TypeScript: remove fallback for `should` to increase type safety
### Current behavior:
```ts
cy.get('foo').should('whatevergoeshere'); // no error :-(

cy.get('foo').should('exis'); // oops, typo—but no error :-(
cy.get('foo').should('exist'); // ✅
```

### Desired behavior:
```ts
cy.get('foo').should('whatevergoeshere'); // type error, `whatevergoeshere` is not a valid assertion

cy.get('foo').should('exis'); // type error, `exis` is not a valid assertion
cy.get('foo').should('exist'); // ✅
```

┆Issue is synchronized with this [Jira Features](https://cypress-io.atlassian.net/browse/TR-161) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by samijaber**
Closes #5573

### User facing changelog

- Removes fallback types for `.should()`, which were accepting any parameters. This means that bad input will now show an error

### How has the user experience changed?

Better dev UX:

image

### CODE REVIEW

File: cli/types/index.d.ts [hunk 1]
```diff
@@ -4090,15 +4090,6 @@ declare namespace Cypress {
...
     (chainer: 'not.match', value: string): Chainable<Subject>
 
-    // fallback
-    /**
-     * Create an assertion. Assertions are automatically retried until they pass or time out.
-     * Ctrl+Space will invoke auto-complete in most editors.
-     * @see https://on.cypress.io/should
-     */
-    (chainers: string, value?: any): Chainable<Subject>
-    (chainers: string, value: any, match: any): Chainable<Subject>
```
Comment (samijaber) [inline cli/types/index.d.ts]: PS: if a user has custom chainers they want to use, they should be able to explicitly add them by extending the `Chainer` `interface`. There is no need for generic `string`/`any` fallbacks.

File: cli/types/tests/chainer-examples.ts [hunk 2]
```diff
@@ -357,8 +357,6 @@ cy.get('#result').should('not.be.focused')
 cy.get('#result').should('have.focus')
 cy.get('#result').should('not.have.focus')
 
-cy.get('#result').should('be.contain', 'text')
```
Comment (samijaber) [inline cli/types/tests/chainer-examples.ts]: There's no `be.contain` as far as I know, so I'm not sure what this is actually meant to test.

Comment (jennifer-shehane) [reply cli/types/tests/chainer-examples.ts]: `be` is just a chainable getter, from our Docs https://docs.cypress.io/guides/references/assertions.html#BDD-Assertions

>These getters are also available for BDD assertions. They don’t actually do anything, but they enable you to write clear, english sentences.

So, removing this is actually removing a valid test

```suggestion
cy.get('#result').should('contain', 'text')
```

**Comment by jennifer-shehane** [comment]
Hey @samijaber, will you have time to address the changes asked of this PR? I will be setting the title to `[WIP]`, please remove WIP when the PR is ready for rereview.

**Comment by samijaber** [comment]
> Hey @samijaber, will you have time to address the changes asked of this PR? I will be setting the title to `[WIP]`, please remove WIP when the PR is ready for rereview.

Hey @jennifer-shehane, yes I am planning to address the changes soon: life slowed down during the holidays. 😄 

I will change the title back once it's ready, thanks!

**Comment by jennifer-shehane** [comment]
@samijaber Hey @samijaber, will you have time to address the changes asked of this PR? We will have to close due to inactivity otherwise.

**Comment by jennifer-shehane** [comment]
Unfortunately we have to close this PR due to inactivity. Please open a new PR addressing the original issue and any requested changes.

**Pull request closed without merging**